openedx/features/content_type_gating/services.py

```diff
@@ -29,4 +46,40 @@ def content_type_gate_for_block(self, user, block, course_id):
             access = has_access(user, 'load', block, course_id)
             if (not access and access.error_code == 'incorrect_user_group'):
                 return access.user_fragment
+
+            # Handle case of staff users masquerading as Learner in Audit or Learner in Limited Access
+            if self._is_masquerading_as_audit_or_limited_access(user, course_id):
+                # Check merged_group_access in case the course steam overrode content type gating
+                # for this problem with the studio override
+                if LIMITED_ACCESS.id not in block.merged_group_access.get(CONTENT_GATING_PARTITION_ID):
+                    return create_content_gating_partition(course).access_denied_fragment(
+                        course, user, LIMITED_ACCESS, [FULL_ACCESS]
+                    )
             return None
+
+    def check_children_for_content_type_gating_paywall(self, item, course, course_id):
+        """
+        Arguments:
+            item (xblock such as a sequence or vertical block)
+            course (xblock.internal.CourseDescriptorWithMixins)
+            course_id (CourseLocator)
+
+        Returns:
+            If:
+            This xblock contains problems which this user cannot load due to content type gating
+            Then:
+            Return the first content type gating paywall (Fragment)
+            Else:
+            Return None
```

Comment:
I've not seen this format for the `Returns` block. Do we actually have anything that parses these things and puts them in readthedocs or somewhere?

Reply:
I was just formatting this to try to make it easily human readable, didn't realize this would be automatically parsed by anything. 
should I be using a different format or not using the Returns: block?

Reply:
There are different parsers available I think - I don't think python requires a specific format. And we may not be running any of them? But it's possible for parsers to read docstrings like this and generate readthedocs content or similar.

I feel like platform usage is varied enough that we probably aren't running a parser. And I don't think I've seen a readthedocs for inner platform code. We certainly don't seem to be linting the format.

Reply:
I removed the returns so that it doesn't look like a specially formatted string